Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -10,10 +14,68 @@ namespace JokesOnYou.Web.Api.Services
     public class JokesService : IJokesService
     {
         private readonly IJokesRepository _jokesRepo;
+        private readonly IUserRepository _userRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
 
-        public JokesService(IJokesRepository jokesRepo)
+        public JokesService(IJokesRepository jokesRepo, IUserRepository userRepository, IUnitOfWork unitOfWork, IMapper mapper)
         {
             _jokesRepo = jokesRepo;
+            _mapper = mapper;
+            _userRepository = userRepository;
+            _unitOfWork = unitOfWork;
+        }
+        public async Task<JokeReplyDto> CreateJokeAsync(JokeCreateDto jokeCreateDto, string userId)
+        {
+            jokeCreateDto.Premise = jokeCreateDto.Premise.Trim();
+            jokeCreateDto.Punchline = jokeCreateDto.Punchline.Trim();
+
+            var isDuplicate = await IsJokeDuplicate(jokeCreateDto);
+            if (isDuplicate)
+            {
+                throw new AppException("Joke Already exists.");
+            }
+
+            var user = await _userRepository.GetUserAsync(userId);
+            var joke = _mapper.Map<Joke>(jokeCreateDto);
+            joke.UploadDate = DateTime.Now;
+            joke.Author = user;
+
+            await _jokesRepo.CreateJokeAsync(joke);
+            var saved = await _unitOfWork.SaveAsync();
+            if (!saved)
+            {
+                throw new AppException("Error with saving the Joke.");
+            }
+
+            var jokeReplyDto = _mapper.Map<JokeReplyDto>(joke);
+
+            return jokeReplyDto;
+        }
+
+        private async Task<bool> IsJokeDuplicate(JokeCreateDto jokeCreateDto)
+        {
+            var jokes = await _jokesRepo.GetJokesByPremiseAsync(jokeCreateDto.Premise);
```

Comment:
Might wanna refactor this a bit. This is something that belongs to the repository. In there i would also just pass down the  punchline since thats all your checking, and i wouldnt wanna load the items to memory. So you can just do something like

return _context.jokes.any(x => x.punchline == punchline)